kavm/src/kavm/kavm.py

```diff
@@ -103,26 +111,45 @@ def run_avm_json(
             tmp_scenario_file.flush()
             _LOGGER.info('Running KAVM')
             os.environ['KAVM_DEFINITION_DIR'] = str(self.definition_dir)
+            print(os.environ['KAVM_DEFINITION_DIR'])
 
             try:
-                proc_result = _krun(
-                    input_file=Path(tmp_scenario_file.name),
-                    definition_dir=self.definition_dir,
-                    output=KRunOutput.KORE,
-                    depth=depth,
-                    no_expand_macros=False,
-                    profile=profile,
-                    check=check,
-                    cmap={'TEAL_PROGRAMS': tmp_teals_file.name},
-                    pmap={'TEAL_PROGRAMS': str(self._catcat_parser)},
-                    pipe_stderr=True,
-                )
-                if proc_result.returncode != 0:
-                    raise RuntimeError('Non-zero exit-code from krun.')
-
-                parser = KoreParser(proc_result.stdout)
-                final_pattern = parser.pattern()
-                assert parser.eof
+                if output == "kore":
+                    proc_result = _krun(
+                        input_file=Path(tmp_scenario_file.name),
+                        definition_dir=self.definition_dir,
+                        output=KRunOutput.KORE,
+                        depth=depth,
+                        no_expand_macros=False,
+                        profile=profile,
+                        check=check,
+                        cmap={'TEAL_PROGRAMS': tmp_teals_file.name},
+                        pmap={'TEAL_PROGRAMS': str(self._catcat_parser)},
+                        pipe_stderr=True,
+                    )
+                    #                if proc_result.returncode != 0:
+                    #                    raise RuntimeError('Non-zero exit-code from krun.')
+
+                    parser = KoreParser(proc_result.stdout)
+                    final_pattern = parser.pattern()
+                    assert parser.eof
+                if output == "pretty":
+                    proc_result = _krun(
+                        input_file=Path(tmp_scenario_file.name),
+                        definition_dir=self.definition_dir,
+                        output=KRunOutput.PRETTY,
+                        depth=depth,
+                        no_expand_macros=False,
+                        profile=profile,
+                        check=check,
+                        cmap={'TEAL_PROGRAMS': tmp_teals_file.name},
+                        pmap={'TEAL_PROGRAMS': str(self._catcat_parser)},
+                        pipe_stderr=True,
+                    )
+                    #                if proc_result.returncode != 0:
+                    #                    raise RuntimeError('Non-zero exit-code from krun.')
```

Comment:
This was again something I introduced, but it could be improved like this maybe. The other thing that was concerning me is that this section is a bit confusing to use in general because as I understand it the same check of whether the return code is zero is being done in `_krun` depending on the value of `check`, and also done here. In addition, the rerun_on_error option isn't currently being used but I think needs to be fixed because right now it just runs but doesn't print the result. Maybe we should just figure out a way to simplify the `run_avm_json` options.
```suggestion
                if output == "kore":
                    output_type = KRunOutput.KORE
                elif output == "pretty":
                    output_type = KRunOutput.PRETTY
                else:
                    raise RuntimeError('Unsupported output type')
                    
                proc_result = _krun(
                    input_file=Path(tmp_scenario_file.name),
                    definition_dir=self.definition_dir,
                    output=output_type,
                    depth=depth,
                    no_expand_macros=False,
                    profile=profile,
                    check=check,
                    cmap={'TEAL_PROGRAMS': tmp_teals_file.name},
                    pmap={'TEAL_PROGRAMS': str(self._catcat_parser)},
                    pipe_stderr=True,
                )
                if proc_result.returncode != 0: 
                    raise RuntimeError('non-zero exit-code from krun.')
```